app/providers/atmosphere/optimization_strategy/default.rb

```diff
@@ -105,7 +120,7 @@ def select
           best_tenant = nil
           best_flavor = nil
           instantiation_cost = Float::INFINITY
-          tmpls.each do |tmpl|
+          @tmpls.each do |tmpl|
```

Comment:
You are right `@tmpl` should be `@tmpls` in my original comment - sorry for it - but the comment is still valid.